app/js/lib/server/settings.js

```diff
@@ -0,0 +1,236 @@
+'use strict';
+
+import EventDispatcher from '../common/event-dispatcher';
+
+// Prefix all entries to avoid collisions.
+const PREFIX = 'calendar-';
+
+const ORIGIN = 'https://calendar.knilxof.org';
+
+/**
+ * API version to use (currently not configurable).
+ * @type {number}
+ * @const
+ */
+const API_VERSION = 1;
+
+/**
+ * Regex to match upper case literals.
+ * @type {RegExp}
+ * @const
+ */
+const UPPER_CASE_REGEX = /([A-Z])/g;
+
+const p = Object.freeze({
+  values: Symbol('values'),
+  storage: Symbol('storage'),
+
+  // Private methods.
+  updateSetting: Symbol('updateSetting'),
+  stringToSettingTypedValue: Symbol('stringToSettingTypedValue'),
+  getDefaultSettingValue: Symbol('getDefaultSettingValue'),
+  onStorage: Symbol('onStorage'),
+});
+
+// Definition of all available settings and their default values (if needed).
+const settings = Object.freeze({
+  // String settings.
+  SESSION: Object.freeze({ key: 'session' }),
```

Comment:
Yes, each setting has a key and can also have a type (boolean, number or string) and a default value (currently unused here).
Look at https://github.com/fxbox/app/blob/master/app/js/lib/foxbox/settings.js#L43 for a more complex example.